dask_expr/concat.py

```diff
@@ -84,8 +84,27 @@ def _simplify_down(self):
         )
 
     def _simplify_up(self, parent):
-        # TODO: implement passing projections through
-        return
+        if isinstance(parent, Projection):
+            columns = parent.columns
+            columns_frame = [
+                sorted(set(frame.columns).intersection(columns))
+                for frame in self._frames
+            ]
+            if all(
+                cols == sorted(frame.columns)
+                for frame, cols in zip(self._frames, columns_frame)
+            ):
+                return
+
+            frames = [
+                frame[cols]
```

Comment:
Can't this give us the wrong number of frames now?

Reply:
Yes sorry, to reliant on tests. Fixed now